src/hedera/hederaEngine.js

```diff
@@ -0,0 +1,404 @@
+          this.walletLocalData.otherData.activationStatusUrl =
+            response.status_url
+          this.addToLoop('checkAccountCreationStatus', 5000)
+        } else {
+          this.log('ErrorInvalidActivationPayment', response.error)
+          throw new Error('ErrorInvalidActivationPayment')
+        }
+      }
+    }
+  }
+
+  async checkAccountCreationStatus(): Promise<void> {
+    const statusUrl = this.walletLocalData.otherData.activationStatusUrl
+
+    if (!statusUrl || this.walletLocalData.otherData.hederaAccount) {
+      return
+    }
+
+    this.log('checkAccountCreationStatus', statusUrl)
+
+    let response
+
+    try {
+      response = await (await this.io.fetch(statusUrl)).json()
+    } catch (e) {
+      this.log('error checking Hedera account creation status', e)
+      return
+    }
+
+    if (response.status === 'error') {
+      delete this.timers.checkAccountCreationStatus
+      throw new Error({
+        message: 'ErrorAccountActivation',
+        reason: response.message
+      })
+    }
+
+    if (response.status === 'success') {
+      this.walletLocalData.otherData.hederaAccount = response.account_id
+      this.walletLocalDataDirty = true
+      this.accountId = new hedera.AccountId(response.account_id)
+      await this.saveWalletLoop()
+
+      delete this.timers.checkAccountCreationStatus
+    }
+  }
+
+  // ****************************************************************************
+  // Public methods
+  // ****************************************************************************
+
+  async loadEngine(
+    plugin: EdgeCurrencyTools,
+    walletInfo: EdgeWalletInfo,
+    opts: EdgeCurrencyEngineOptions
+  ): Promise<void> {
+    await super.loadEngine(plugin, walletInfo, opts)
+
+    if (!this.walletLocalData.otherData.hederaAccount) {
+      const accountId = await this.getAccountId()
+
+      if (accountId) {
+        this.accountId = accountId
+        this.walletLocalData.otherData.hederaAccount = accountId.toString()
+        this.walletLocalDataDirty = true
+      } else {
+        await this.checkAccountCreationStatus()
+      }
+    } else {
+      this.accountId = new hedera.AccountId(
+        this.walletLocalData.otherData.hederaAccount
+      )
+    }
+  }
+
+  async startEngine() {
+    console.log('hederaEngine/startEngine', this)
+    this.engineOn = true
+
+    // function is a no-op if the account ID is not yet available
+    this.addToLoop('updateBalance', 5000)
+
+    if (!this.walletLocalData.otherData.hederaAccount) {
+      const accountId = await this.getAccountId()
+
+      if (accountId) {
+        this.accountId = accountId
+        this.walletLocalData.otherData.hederaAccount = accountId.toString()
+        this.walletLocalDataDirty = true
+      } else {
+        this.addToLoop('checkAccountCreationStatus', 5000)
+      }
+    } else {
+      this.accountId = new hedera.AccountId(
+        this.walletLocalData.otherData.hederaAccount
+      )
+    }
```

Comment:
This is duplicated with the above because I'm not sure where the most appropriate place to do this check is.

Reply:
Sorry, I dont quite understand what you are referring to here. It might be good to get on a call to discuss further.

Reply:
This block at lines 203-217 is duplicated between `startEngine()` and `loadEngine()` because I wasn't sure where the best place to have this logic is and opted for redundancy rather than making the wrong choice.

Reply:
`startEngine()` only